x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/ListEnginesAction.java

```diff
@@ -0,0 +1,140 @@
+import org.elasticsearch.xpack.core.action.util.PageParams;
+import org.elasticsearch.xpack.core.action.util.QueryPage;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class ListEnginesAction extends ActionType<ListEnginesAction.Response> {
+
+    public static final ListEnginesAction INSTANCE = new ListEnginesAction();
+    public static final String NAME = "cluster:admin/engine/list";
+
+    public ListEnginesAction() {
+        super(NAME, ListEnginesAction.Response::new);
+    }
+
+    // TODO Check if this should implement IndicesRequest.Replaceable
+    public static class Request extends ActionRequest {
+
+        private static final String DEFAULT_QUERY = "*";
+        private final String query;
+        private final PageParams pageParams;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.query = Objects.requireNonNullElse(in.readOptionalString(), DEFAULT_QUERY);
+            this.pageParams = new PageParams(in);
+        }
+
+        public Request(@Nullable String query, PageParams pageParams) {
+            this.query = Objects.requireNonNullElse(query, DEFAULT_QUERY);
+            this.pageParams = pageParams;
+        }
+
+        public String query() {
+            return query;
+        }
+
+        public PageParams pageParams() {
+            return pageParams;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            // Pagination validation is done as part of PageParams constructor
+            ActionRequestValidationException validationException = null;
+            if (Strings.isEmpty(query())) {
+                validationException = ValidateActions.addValidationError("Engine query is missing", validationException);
+            }
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeOptionalString(query);
+            pageParams.writeTo(out);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return Objects.equals(query, that.query) && Objects.equals(pageParams, that.pageParams);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(query, pageParams);
+        }
+    }
+
+    public static class Response extends ActionResponse implements StatusToXContentObject {
+
+        public static final ParseField ENGINES_RESULT_FIELD = new ParseField("results");
+
+        final QueryPage<Engine> queryPage;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.queryPage = new QueryPage<>(in, Engine::new);
```

Comment:
Used QueryPage in the response per Jim's suggestion. It makes sense to use other Elasticsearch constructs, and making responses in line with other ES APIs instead of adding `_meta` like Ent Search did